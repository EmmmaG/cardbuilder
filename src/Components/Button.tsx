import * as React from 'react'
import { withStyles, WithStyles, Theme } from '@material-ui/core/styles'
import ButtonComponent from '@material-ui/core/Button'

const styles = (theme: Theme) => ({
	button: {
		color: 'white',
        backgroundColor: theme.palette.grey[500],
        fontSize: '20px',
		fontWeight: 450,
		textTransform: 'none' as 'none',
		padding: '3px 20px',
        width: '200px',
		minWidth: '200px',
		border: 'none',
        borderRadius: '3px',
	},
})

export interface Props extends WithStyles<typeof styles> {
    buttonContent: string
    onClick: () => void
    backgroundColor?: string
}

export class Button extends React.Component<Props> {
	public render() {
		const { classes, buttonContent, onClick, backgroundColor } = this.props

		return (
			<ButtonComponent
                variant='contained'
				className={classes.button}
                onClick={onClick}
                style={{backgroundColor}}
			>
				{buttonContent}
			</ButtonComponent>
		)
	}
}

export default withStyles(styles)(Button)
