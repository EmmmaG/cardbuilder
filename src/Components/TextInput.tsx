import * as React from 'react'
import { withStyles, WithStyles, Theme } from '@material-ui/core/styles'
import Select from '@material-ui/core/Select'
import Typography from '@material-ui/core/Typography'
import TextField from '@material-ui/core/TextField'

const styles = (theme: Theme) => ({
	inputContainer: {
		textAlign: 'left' as 'left',
		marginTop: '5px',
		marginBottom: '5px',
		marginLeft: '10px',
		marginRight: '10px',
		display: 'inline-block',
	},
	textField: {
		height: '35px',
		backgroundColor: 'white',
		width: '195px',
	},
	inputRoot: {
		height: '35px',
		fontFamily: '"Big Caslon", "Times New Roman", "serif"',
	},
	title: {
		color: theme.palette.grey[700],
		marginBottom: '2px',
	},
})

export interface Select {
	value: string
	text: string
}

export interface Props extends WithStyles<typeof styles> {
	title: string
	onChange: (selectedValue: string) => void
}

export interface State {
    value: string
}

export class TextInput extends React.Component<Props, State> {
    public state = { value: ''}

	private handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        const value = event.target.value

        this.setState({ value }, () => this.props.onChange(value))
	}

	public render() {
        const { classes, title } = this.props
        const { value } = this.state

		return (
			<div className={classes.inputContainer}>
				<Typography className={classes.title}>{title}</Typography>
				<TextField
					id={title}
					className={classes.textField}
					value={value}
					onChange={this.handleChange}
					variant='outlined'
					InputProps={{ classes: { root: classes.inputRoot } }}
				/>
			</div>
		)
	}
}

export default withStyles(styles)(TextInput)
