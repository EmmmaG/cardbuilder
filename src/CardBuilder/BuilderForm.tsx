import * as React from 'react'
import { withStyles, WithStyles, Theme } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'

import TextInput from '../Components/TextInput'
import Button from '../Components/Button'
import Divider from '../Components/Divider'


const styles = (theme: Theme) => ({
    title: {
        fontSize: '32px',
        fontWeight: 900,
        color: '#374D60',
        marginLeft: '10px',
        textAlign: 'left' as 'left',
    },
    mainContainer: {
        backgroundColor: 'white',
        maxWidth: '430px',
        textAlign: 'center' as 'center',
        padding: '30px',
        display: 'inline-block',
    },
    buttonContainer: {
        marginTop: '30px',
    },
})

export interface FormValue {
    givenName: string
    surname: string
    email: string
    phone: string
    houseName: string
    street: string
    suburb: string
    state: string
    postcode: string
    country: string
}

export interface Props extends WithStyles<typeof styles> {
    onFormChange: (formValue: FormValue) => void
}

export interface State {
    formValue: FormValue
}

export const emptyFormValue: FormValue = {
    givenName: '',
    surname: '',
    email: '',
    phone: '',
    houseName: '',
    street: '',
    suburb: '',
    state: '',
    postcode: '',
    country: '',
}

type FormKey = keyof FormValue

export class BuilderForm extends React.Component<Props, State> {
    public constructor(props: Props) {
        super(props)

        this.state = {
            formValue: emptyFormValue,
        }
    }

    private onTextFieldChange = (value: string, formKey: FormKey) => {
        const currentFormValue = this.state.formValue

        this.setState({
            formValue: {
                ...currentFormValue,
                [formKey]: value,
            }
        }, () => {this.props.onFormChange(this.state.formValue)})
    }

    private renderTextInput = (title: string, formKey: FormKey) => {
        return (
            <TextInput key={formKey} title={title} onChange={(value) => this.onTextFieldChange(value, formKey)} />
        )
    }

    private renderPersonalDetails = () => {
        const nameRow = [{
            title: 'GIVEN NAME',
            formKey: 'givenName' as FormKey,
        }, {
            title: 'SURNAME',
            formKey: 'surname' as FormKey,
        }].map(element => this.renderTextInput(element.title, element.formKey))

        const detailRow = [{
            title: 'EMAIL',
            formKey: 'email' as FormKey,
        }, {
            title: 'PHONE',
            formKey: 'phone' as FormKey,
        }].map(element => this.renderTextInput(element.title, element.formKey))

        return (
            <div>
                <Divider title='PERSONAL DETAILS' />
                <div>
                    {nameRow}
                </div>
                <div>
                    {detailRow}
                </div>
            </div>
        )
    }

    private renderAddress = () => {
        const streetRow = [{
            title: 'HOUSE NAME OR #',
            formKey: 'houseName' as FormKey,
        }, {
            title: 'STREET',
            formKey: 'street' as FormKey,
        }].map(element => this.renderTextInput(element.title, element.formKey))

        const suburbRow = [{
            title: 'SUBURB',
            formKey: 'suburb' as FormKey,
        }, {
            title: 'STATE',
            formKey: 'state' as FormKey,
        }].map(element => this.renderTextInput(element.title, element.formKey))

        const postcodeRow = [{
            title: 'POSTCODE',
            formKey: 'postcode' as FormKey,
        }, {
            title: 'COUNTRY',
            formKey: 'country' as FormKey,
        }].map(element => this.renderTextInput(element.title, element.formKey))

        return (
            <div>
                <Divider title='PERSONAL DETAILS' />
                <div>
                    {streetRow}
                </div>
                <div>
                    {suburbRow}
                </div>
                <div>
                    {postcodeRow}
                </div>
            </div>
        )
    }

    private renderButtons = () => (
        <div className={this.props.classes.buttonContainer}>
            <Button
                buttonContent='Upload Avatar'
                onClick={() => console.log('a')}
                backgroundColor='#768E9C'
            />
            <Button
                buttonContent='Create hCard'
                onClick={() => console.log('a')}
                backgroundColor='#43A8E0'
            />
        </div>
    )

    public render() {
        const { classes } = this.props

        return (
            <div className={classes.mainContainer}>
				<Typography className={classes.title}>hCard Builder</Typography>

                {this.renderPersonalDetails()}

                {this.renderAddress()}

                {this.renderButtons()}
            </div>
        )
    }
}

export default withStyles(styles)(BuilderForm)
