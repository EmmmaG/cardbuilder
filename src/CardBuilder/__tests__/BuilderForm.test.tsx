import * as React from 'react'
import { shallow } from 'enzyme'
import idObj from 'identity-obj-proxy'
import Button from '../../Components/Button'
import TextInput from '../../Components/TextInput'
import BuilderForm, { Props } from '../BuilderForm'

const props: Props = {
    onFormChange: jest.fn(),
    onAvatarSelected: jest.fn(),
	classes: idObj,
}

describe('BuilderForm', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<BuilderForm {...props} />).dive()
		expect(wrapper).toMatchSnapshot()
    })

    it('should update state with correct input value', () => {
		const wrapper = shallow(<BuilderForm {...props} />).dive()

        const textInput = wrapper.find(TextInput).findWhere(t => t.prop('title') === 'GIVEN NAME')
        const value = 'jon'
        textInput.prop('onChange')(value)

        expect(wrapper.state('formValue')).toEqual(expect.objectContaining({ givenName: value }))
    })

    it('should not open alert dialog if email is valid', () => {
        const wrapper = shallow(<BuilderForm {...props} />).dive()
        const emailInput = wrapper.find(TextInput).findWhere(t => t.prop('title') === 'EMAIL')

        const validEmail = 'jon@gmail.com'
        emailInput.prop('onChange')(validEmail)

        const createButton = wrapper.find(Button).findWhere(b => b.prop('buttonContent') === 'Create hCard')

        createButton.simulate('click')

        expect(wrapper.state('isOpen')).toBeFalsy()
    })

    it('should open alert dialog if email is not valid', () => {
        const wrapper = shallow(<BuilderForm {...props} />).dive()
        const emailInput = wrapper.find(TextInput).findWhere(t => t.prop('title') === 'EMAIL')

        const invalidEmail = 'jon@A'
        emailInput.prop('onChange')(invalidEmail)

        const createButton = wrapper.find(Button).findWhere(b => b.prop('buttonContent') === 'Create hCard')

        createButton.simulate('click')

        expect(wrapper.state('isOpen')).toBeTruthy()
    })
})
