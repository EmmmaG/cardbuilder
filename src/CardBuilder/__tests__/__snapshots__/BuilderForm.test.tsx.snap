// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuilderForm renders correctly 1`] = `
<Fragment>
  <div
    className="BuilderForm-mainContainer-2"
  >
    <WithStyles(Typography)
      className="BuilderForm-title-1"
    >
      hCard Builder
    </WithStyles(Typography)>
    <div>
      <WithStyles(Divider)
        title="PERSONAL DETAILS"
      />
      <div>
        <WithStyles(TextInput)
          key="givenName"
          onChange={[Function]}
          title="GIVEN NAME"
        />
        <WithStyles(TextInput)
          key="surname"
          onChange={[Function]}
          title="SURNAME"
        />
      </div>
      <div>
        <WithStyles(TextInput)
          key="email"
          onChange={[Function]}
          title="EMAIL"
        />
        <WithStyles(TextInput)
          key="phone"
          onChange={[Function]}
          title="PHONE"
        />
      </div>
    </div>
    <div>
      <WithStyles(Divider)
        title="PERSONAL DETAILS"
      />
      <div>
        <WithStyles(TextInput)
          key="houseName"
          onChange={[Function]}
          title="HOUSE NAME OR #"
        />
        <WithStyles(TextInput)
          key="street"
          onChange={[Function]}
          title="STREET"
        />
      </div>
      <div>
        <WithStyles(TextInput)
          key="suburb"
          onChange={[Function]}
          title="SUBURB"
        />
        <WithStyles(TextInput)
          key="state"
          onChange={[Function]}
          title="STATE"
        />
      </div>
      <div>
        <WithStyles(TextInput)
          key="postcode"
          onChange={[Function]}
          title="POSTCODE"
        />
        <WithStyles(TextInput)
          key="country"
          onChange={[Function]}
          title="COUNTRY"
        />
      </div>
    </div>
    <div
      className="BuilderForm-buttonContainer-3"
    >
      <WithStyles(UploadFileButton)
        onFileLoaded={[Function]}
      />
      <WithStyles(Button)
        backgroundColor="#43A8E0"
        buttonContent="Create hCard"
        onClick={[Function]}
      />
    </div>
  </div>
  <WithStyles(Dialog)
    onClose={[Function]}
    open={false}
  >
    <WithStyles(DialogContent)>
      <WithStyles(DialogContentText)>
        Please use a valid email address.
      </WithStyles(DialogContentText)>
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        buttonContent="Got it!"
        onClick={[Function]}
      />
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</Fragment>
`;
